feat: add automated Node.js version management workflow

salmanmkc opened the pull request:
## Overview
This PR adds automated Node.js version management to keep the runner current with the latest LTS versions.

## What's Added

### 🔄 Node.js Version Upgrade Workflow (`.github/workflows/node-upgrade.yml`)
- **Automated version checking** for Node 20.x and 24.x LTS releases
- **Smart updates** to `NODE20_VERSION` and `NODE24_VERSION` in `src/Misc/externals.sh`
- **Weekly schedule** (Mondays at 6 AM) plus manual triggers
- **PR creation** only when newer versions are available

## Key Features
- ✅ **Dual Node version support** (Node 20 LTS + Node 24 LTS)
- ✅ **Official version source** (actions/node-versions manifest)
- ✅ **Smart change detection** (only creates PR if versions differ)
- ✅ **Proper automation** (git operations with --no-verify)
- ✅ **Alpine reminder** (manual alpine_nodejs release follow-up)

## Dependencies
- **Requires**: Labels from actions/runner#4024 (`dependency` label)
- **Integrates with**: Monitoring from actions/runner#4025

## Testing
- ✅ Tested version detection logic
- ✅ Verified externals.sh update mechanism  
- ✅ Confirmed PR creation with proper formatting

## Next Steps
- This workflow handles Node.js runtime versions
- Separate PR will handle npm security management (npm-audit workflows)

This ensures the runner stays current with supported Node.js versions while maintaining dual-version compatibility for GitHub Actions.

salmanmkc committed:
feat: add automated Node.js version management workflow

## Node.js Version Upgrade Workflow
- Automatically checks for latest Node 20.x and 24.x versions
- Updates NODE20_VERSION and NODE24_VERSION in src/Misc/externals.sh
- Creates PRs when newer versions are available
- Weekly schedule (Mondays at 6 AM) plus manual triggers

## Key Features
- ✅ Dual Node version support (Node 20 LTS + Node 24 LTS)
- ✅ Fetches versions from actions/node-versions manifest
- ✅ Smart change detection (only creates PR if versions differ)
- ✅ Proper git operations with --no-verify for automated commits
- ✅ Includes alpine_nodejs release reminder for manual follow-up

## Dependencies
- Requires dependency labels from actions/runner#4024
- Works with monitoring from actions/runner#4025

This workflow ensures the runner stays current with supported Node.js versions while maintaining dual-version compatibility.

TingluoHuang commented on .github/workflows/node-upgrade.yml:
```diff
@@ -0,0 +1,92 @@
+        id: node-versions
+        run: |
+          # Get latest versions from the versions manifest
+          VERSIONS_JSON=$(curl -s https://raw.githubusercontent.com/actions/node-versions/main/versions-manifest.json)
```
I think we should use `https://github.com/actions/alpine_nodejs/releases` and `https://github.com/nodejs/node/releases` to help decide which version is good for us.

salmanmkc committed:
feat: improve Node.js version detection using official sources

## Enhanced Version Detection Strategy
- **Primary source**: Official Node.js GitHub releases API
- **Compatibility check**: Verify versions exist in alpine_nodejs releases
- **Smart fallback**: Use latest available alpine version if official version not ready

## Key Improvements
- ✅ More reliable than third-party versions manifest
- ✅ Ensures alpine_nodejs compatibility before updating
- ✅ Prevents workflow failures from version mismatches
- ✅ Better logging and error handling
- ✅ Uses official GitHub APIs for both Node.js and alpine_nodejs

## Sources Used
- Node.js releases: https://github.com/nodejs/node/releases
- Alpine compatibility: https://github.com/actions/alpine_nodejs/releases

This approach provides better control and reliability for Node.js version management while ensuring container compatibility.

TingluoHuang commented on .github/workflows/node-upgrade.yml:
```diff
@@ -0,0 +1,118 @@
+          ALPINE_RELEASES=$(curl -s https://api.github.com/repos/actions/alpine_nodejs/releases | jq -r '.[].tag_name')
+          
+          if ! echo "$ALPINE_RELEASES" | grep -q "^node20-$LATEST_NODE20$"; then
+            echo "Warning: Node 20 version $LATEST_NODE20 not found in alpine_nodejs releases"
```
should we use the warning command?
ex:
`echo "::warning file=app.js,line=1,col=5,endColumn=7,title=YOUR-TITLE::Missing semicolon"`

salmanmkc committed:
Add echo warning

salmanmkc commented on .github/workflows/node-upgrade.yml:
```diff
@@ -0,0 +1,92 @@
+          # Create branch and commit changes
+          branch_name="chore/update-node"
+          git checkout -b "$branch_name"
+          git commit -a -m "chore: update Node versions (20: ${{ steps.node-versions.outputs.latest_node20 }}, 24: ${{ steps.node-versions.outputs.latest_node24 }})"
```
not true
